### PR TITLE
Add in changes to fix connections not being closed properly

### PR DESCRIPTION
**Pull request by tamimkh**
Backporting https://github.com/brianmario/mysql2/pull/663 and https://github.com/brianmario/mysql2/pull/702

### CODE REVIEW

**Comment by tamimkh** [comment]
Reviewers please don't merge this. We want this to be a separate branch

File: ext/mysql2/client.c [hunk 1]
```diff
@@ -222,27 +222,14 @@ static VALUE invalidate_fd(int clientfd)
 #endif /* _WIN32 */
 
 static void *nogvl_close(void *ptr) {
-  mysql_client_wrapper *wrapper;
-  wrapper = ptr;
-  if (wrapper->connected) {
-    wrapper->active_thread = Qnil;
-    wrapper->connected = 0;
-#ifndef _WIN32
```
Comment (burnison) [inline ext/mysql2/client.c]: Did upstream also remove the call to `invalidate_fd` in this cherry-picked commit?

Comment (burnison) [reply ext/mysql2/client.c]: On second thought, I see it got moved. Is `decr_mysql2_client` invoked in the normal flow of this call? Without this function being static, I'd be a bit concerned about leaks.

**Comment by tamimkh** [comment]
Was merged and being used currently in web. Going to close PR though since we still need the branch for prosterity

**Pull request closed without merging**